### PR TITLE
Updates PLS to reflect changes in the AbcSmc repo & makes Eigen a submodule

### PR DESCRIPTION
**Pull request by pearsonca**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+As a general rule, we don’t assign issues to anyone. If you find an issue to work on, you are welcome to open a PR with a fix.
+
+### Make Changes
+
+#### Make changes in the UI
+
+Click **Make a contribution** at the bottom of any docs page to make small changes such as a typo, sentence fix, or a broken link. This takes you to the `.md` file where you can make your changes and [create a pull request](#pull-request) for a review.
+
+ <img src="/contributing/images/contribution_cta.png" />
```
Comment (tjhladish) [inline CONTRIBUTING.md:58]: Make this accessable

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+add_library(pls ${CMAKE_CURRENT_SOURCE_DIR}/src/pls.cpp)
+add_library(${CMAKE_PROJECT_NAME}::pls ALIAS pls)
+
+set_target_properties(pls PROPERTIES
+                      VERSION ${${CMAKE_PROJECT_NAME}_VERSION}
+                      SOVERSION ${${CMAKE_PROJECT_NAME}_VERSION_MAJOR})
+
+target_include_directories(pls PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>")
+
+target_compile_features(pls PUBLIC cxx_std_20)
```
Comment (tjhladish) [inline CMakeLists.txt:32]: Preference to keep this to 17, if that is reasonable

File: Makefile [hunk 3]
```diff
@@ -1,16 +1,17 @@
-CFLAGS=--ansi --pedantic -std=c++0x -O2
-#CFLAGS=--ansi --pedantic -std=c++0x -g 
-INCLUDE= -I.
-all: pls
 
-pls: pls.cpp 
-	g++ $(CFLAGS) $(INCLUDE) pls.cpp utility.cpp -o pls
+.FORCE:
 
-pls_mpreal: pls.cpp
```
Comment (tjhladish) [inline Makefile:9]: Does this still exist as a thing?  I'm not seeing anything in the new code that looks like a mpreal version.  I'm not asserting that's a problem, just asking.

File: include/PLS/pls.h [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        std::vector<float_type> vec(data.begin(), data.end());
+        return vec;
+    }
+
+    // converts a std::vector to an Eigen vector type (Row or Col)
+    // TODO: should this be a Eigen::Map type (i.e. a Matrix-like view of the data)?
+    template<typename EIGENTYPE>
+    inline EIGENTYPE to_evector(const std::vector<float_type> & data) {
+        EIGENTYPE row(data.size());
+        for (size_t i = 0; i < data.size(); i++) row[i] = data[i];
```
Comment (tjhladish) [inline include/PLS/pls.h:67]: {}

File: include/PLS/pls.h [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    // output methods
+    void print_explained_variance(
+        const Mat2D& X, const Mat2D& Y, std::ostream& os = std::cerr
+    ) const;
+
+    void print_state(std::ostream& os = std::cerr) const;
+
+    private:
+        size_t A; // number of components
+        Mat2Dc P, W, R, Q, T;
```
Comment (tjhladish) [inline include/PLS/pls.h:245]: Are there getters for all of these?

File: src/main.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+#include <random>
+#include <iostream>
+
+#include <PLS/pls.h>
+
+using namespace PLS;
+
+int main(int argc, char *argv[]) {
+
+    if (argc < 4) {
```
Comment (tjhladish) [inline src/main.cpp:12]: argc != 4 maybe?

File: CONTRIBUTING.md (hunk 1 above)
Comment (pearsonca) [reply CONTRIBUTING.md:58]: OBE, deleted in upstream merge

File: CMakeLists.txt (hunk 2 above)
Comment (pearsonca) [reply CMakeLists.txt:32]: done in incoming upstream merge

File: include/PLS/pls.h [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+            [&v](const auto & lhs, const auto & rhs) {
+                return *(v.begin() + lhs) < *(v.begin()+ rhs);
+            }
+        );
+        return result;
+    }
+
+    // converts an Eigen vector type (Row or Col) to a std::vector
+    template<typename EIGENTYPE>
+    std::vector<float_type> to_cvector(const EIGENTYPE & data) {
```
[the remark this answers is not in the document]
Comment (pearsonca) [reply include/PLS/pls.h:57]: not really - they do fancy things with underlying data, encapsulated behind a single interface - the begin/end grab is the exposed-way-that-works.

File: include/PLS/pls.h (hunk 5 above)
Comment (pearsonca) [reply include/PLS/pls.h:245]: no, intentionally.

File: src/main.cpp (hunk 6 above)
Comment (pearsonca) [reply src/main.cpp:12]: sure, will fix once incoming arrives.

**Pull request merged**